### DISCUSSION TITLE
How to wrap text around an image?

### DISCUSSION THREAD

**Discussion by yusancky** [typst/typst#1069] [Q&A] [answered]
I want to make an image display on the right side, but not affect the text content on the left as below screenshot shown.

![Screenshots in the Apple Pages](https://user-images.githubusercontent.com/83524927/235634564-bc7ef125-af8a-43b6-9469-1b2623a6a5fb.jpeg)

Is there any way to do it?

**Comment by laurmaedje** [comment]
There is no automatic way to do this at the moment. The only two options are to insert all line breaks manually or to split up the text into two pieces (you still need to know the exact point where the last short line ends).

**Reply by yusancky** [reply]
Thank you so much! I'm trying it. 

Could you tell me if Typst will implement this feature in the future?

**Reply by laurmaedje** [reply]
Yes, it will

**Comment by snowc0de** [comment, marked as answer]
I reply here because I had the same need and thought maybe others would be interested. I considered the thing was split in 3 parts (the left text part, the right image part and the bottom text)

```typst
#let imageonside(lefttext, rightimage, bottomtext: none, marginleft: 0em, margintop: 0.5em) = {
  set par(justify: true)
  grid(columns: 2, column-gutter: 1em, lefttext, rightimage)
  set par(justify: false)
  block(inset: (left: marginleft, top: -margintop), bottomtext)
}

#imageonside(lorem(100), square(height: 200pt, width: 200pt), bottomtext: lorem(100))
```

The result is this : 

![image](https://github.com/typst/typst/assets/33733474/d0a0f2ae-e36c-4a90-811e-710d9943c44c)

So yeah, you still need to split the text, but it's easier than manually placing break lines. Also, the marginleft and margintop parameters lets you offset the bottom text horizontally or vertically according to your needs (for instance if you have a list, you might need to offset the text to the right)

PS: if you don't want the justify, just delete the `set` in the function.

**Reply by Toniolo-Marco** [reply]
Same issue, your solution is ok but typst will not render correctly the inner text. What I mean is that, for example, the text: `"an equation $a = b$"` won't be rendered as an equation. Is there a solution?

**Reply by snowc0de** [reply]
Make sure to use brackets and not quotes. So it should be `[an equation $a = b$]` instead.

In typst quotes means plain text while brackets mean rich text. So if you use quotes your equations won't be rendered.

**Comment by warpmatrix** [comment]
Update a package for who may be interested: [wrap-it](https://typst.app/universe/package/wrap-it)
